test/Microsoft.AspNet.JsonPatch.Test/NestedObjectTests.cs

```diff
@@ -202,6 +202,151 @@ public void AddToListWithSerialization()
         }
 
         [Fact]
+        public void AddToIntListToGeneralList()
```

Comment:
`JsonProperty`'s `PropertyType` is fetching declared type of the property not the runtime type.